docs(nav-drawer): Add nav-drawer topic (#196)
IgniteUI/igniteui-xplat-docs: doc/en/components/navigation-drawer.md

> ## $Platform$ Navigation Drawer Overview

Reviewer: This should be single #.
```suggestion
# $Platform$ Navigation Drawer Overview
```

            </igc-nav-drawer-item>
        </igc-nav-drawer>
    </div>
```

Let's also add some radio buttons to display all `position` values. This way whenever one gets selected, we will change the position of the drawer.

```ts
// ...
import 'igniteui-webcomponents';
import { IgcNavDrawerComponent } from 'igniteui-webcomponents/src/components/nav-drawer/nav-drawer';
import { IgcRadioGroupComponent } from 'igniteui-webcomponents/src/components/radio-group/radio-group';

this.navDrawer = document.getElementById('navDrawer') as IgcNavDrawerComponent;
this.radioGroup = document.getElementById('radio-group') as IgcRadioGroupComponent;
this.radioGroup.addEventListener('click', (radio: any) => {
    this.navDrawer.position = radio.target.value;
});
```

```html
<igc-radio-group id="radio-group" alignment="horizontal">
    <igc-radio name="position" value="start" label-position="after">Start</igc-radio>
    <igc-radio name="position" value="end" label-position="after">End</igc-radio>
    <igc-radio name="position" value="top" label-position="after" checked="true">Top</igc-radio>
    <igc-radio name="position" value="bottom" label-position="after" checked="true">Bottom</igc-radio>

> </igc-radio-group>

Reviewer: This code snippet looks different than the actual source code.

            <igc-nav-drawer-header-item>Sample Drawer</igc-nav-drawer-header-item>

            <igc-nav-drawer-item>
                <igc-icon slot="icon" name="home"></igc-icon>
                <span slot="content">Home</span>
            </igc-nav-drawer-item>

            <igc-nav-drawer-item>
                <igc-icon slot="icon" name="search"></igc-icon>
                <span slot="content">Search</span>
            </igc-nav-drawer-item>

            <div slot="mini">
                <igc-nav-drawer-item>
                    <igc-icon slot="icon" name="home"></igc-icon>
                </igc-nav-drawer-item>

                <igc-nav-drawer-item>
                    <igc-icon slot="icon" name="search"></igc-icon>
                </igc-nav-drawer-item>
            </div>
        </igc-nav-drawer>
    </div>
```

And here's the result:

<code-view style="height: 500px"
           data-demos-base-url="{environment:dvDemosBaseUrl}"
           iframe-src="{environment:dvDemosBaseUrl}/menus/nav-drawer-add-mini"
           alt="$Platform$ Navigation Drawer Mini Example"
           github-src="menus/nav-drawer/add-mini">
</code-view>

## Styling the Navigation Drawer

> The navigation drawer component exposes several CSS parts (`base`, `main`, and `mini`) to give you full control over its styling.

Reviewer: We should list the CSS parts of the drawer item too.

                <igc-nav-drawer-item>
                    <igc-icon slot="icon" name="home"></igc-icon>
                </igc-nav-drawer-item>

                <igc-nav-drawer-item>
                    <igc-icon slot="icon" name="search"></igc-icon>
                </igc-nav-drawer-item>
            </div>
        </igc-nav-drawer>
    </div>
```

And here's the result:

<code-view style="height: 500px"
           data-demos-base-url="{environment:dvDemosBaseUrl}"
           iframe-src="{environment:dvDemosBaseUrl}/menus/nav-drawer-add-mini"
           alt="$Platform$ Navigation Drawer Mini Example"
           github-src="menus/nav-drawer/add-mini">
</code-view>

## Styling the Navigation Drawer

The navigation drawer component exposes several CSS parts - `base`, `main`, and `mini`. The navigation drawer item component exposes three CSS parts as well - `base`, `icon` and `content`, giving you full control over their styling.

```scss
igc-nav-drawer::part(base) {
    background: #2d313a;
}

igc-nav-drawer-item::part(base) {
    color: #fff;
}

igc-nav-drawer-item[active]::part(base) {
    background: #f3c03e;
    color: #2c2c2c
}

igc-nav-drawer-header-item {
    color: #f3c03e

> }

Reviewer: Update code snippet with the hover style.

# $Platform$ Navigation Drawer Overview

The Infragistics $Platform$ Navigation Drawer is a WebComponent that provides side navigation. It can be expanded or collapsed within the content. A mini version provides quick access to navigation even when closed. Its content is completely customizable while also providing default menu item styling.

> > [!NOTE]

Reviewer 2: Remove this note. We will have to add it in the general getting started topic because it's applicable to all components.

> The Infragistics $Platform$ [IgcNavDrawerComponent]({environment:wcApiUrl}/classes/igcnavdrawercomponent.html) is a WebComponent that provides side navigation. It can be expanded or collapsed within the content. A mini version provides quick access to navigation even when closed. Its content is completely customizable while also providing default menu item styling.

Reviewer: I would suggest to use **Navigation Drawer** here instead the component class name since the first paragraph is very important for the SEO. You can use the component class name in the **Usage** section below.

<code-view style="height: 300px"
           data-demos-base-url="{environment:dvDemosBaseUrl}"
           iframe-src="{environment:dvDemosBaseUrl}/menus/nav-drawer-add-drawer-items"
           alt="$Platform$ Navigation Drawer Items Example"
           github-src="menus/nav-drawer/add-drawer-items">
</code-view>

## Usage

> To get started with the Navigation Drawer Web component, first you need to install Ignite UI for Web Components by executing the following command:

Reviewer: ```suggestion
To get started with the Navigation Drawer Web component, first you need to install $ProductName$ by executing the following command:
```

```cmd

> npm install igniteui-webcomponents

Reviewer: ```suggestion
npm install {PackageComponents}
```

# $Platform$ Navigation Drawer Overview

> The Infragistics $Platform$ Navigation Drawer is a WebComponent that provides side navigation. It can be expanded or collapsed within the content. A mini version provides quick access to navigation even when closed. Its content is completely customizable while also providing default menu item styling.

Reviewer 2: Make one of the first occurences of the component name a link to the class.

```

## Adding Navigation Drawer Items

After the Navigation Drawer component is imported, we are ready to start using it! We can add the following code that displays a simple Navigation Drawer with some items:

```html
    <div style="display: flex;">
        <igc-nav-drawer open="true">
            <igc-nav-drawer-header-item>
                Sample Drawer
            </igc-nav-drawer-header-item>

            <igc-nav-drawer-item>
                <igc-icon slot="icon" name="home"></igc-icon>
                <span slot="content">Home</span>
            </igc-nav-drawer-item>

            <igc-nav-drawer-item>
                <igc-icon slot="icon" name="search"></igc-icon>
                <span slot="content">Search</span>
            </igc-nav-drawer-item>
        </igc-nav-drawer>
    </div>
```

If all went well, you should see the following in your browser:

<code-view style="height: 300px"
           data-demos-base-url="{environment:dvDemosBaseUrl}"
           iframe-src="{environment:dvDemosBaseUrl}/menus/nav-drawer-add-drawer-items"
           alt="$Platform$ Navigation Drawer Items Example"
           github-src="menus/nav-drawer/add-drawer-items">
</code-view>

> While any content can be provided in the drawer, the `igc-nav-drawer-item` is available to apply out-of-the-box styling to the items.

Reviewer: Make the `igc-nav-drawer-item` a link to the API docs.

> To enhance our component a bit, we can use it in conjunction with the `IgcNavbarComponent`. This way we can achieve a more completed look and use the drawer's methods. Let's look at how we can use this in the next example:

Reviewer: `IgcNavbarComponent` should be link.

```html
    <igc-navbar>
      <igc-icon slot="start" name="menu" id="menu"></igc-icon>
      <h2>Home</h2>
    </igc-navbar>

    <div id="root" style="display: flex;">
        <igc-nav-drawer id="navDrawer">
            <igc-nav-drawer-header-item>
                Sample Drawer
            </igc-nav-drawer-header-item>

            <igc-nav-drawer-item>
                <igc-icon slot="icon" name="home"></igc-icon>
                <span slot="content">Home</span>
            </igc-nav-drawer-item>

            <igc-nav-drawer-item>
                <igc-icon slot="icon" name="search"></igc-icon>
                <span slot="content">Search</span>
            </igc-nav-drawer-item>
        </igc-nav-drawer>
    </div>
```

> Let's also add some radio buttons to display all `position` values. This way whenever one gets selected, we will change the position of the drawer.

Reviewer: `position` should be link.

}

igc-nav-drawer-item[active]::part(base) {
    background: #f3c03e;
    color: #2c2c2c
}

igc-nav-drawer-header-item {
    color: #f3c03e
}
```

<code-view style="height: 500px"
           data-demos-base-url="{environment:dvDemosBaseUrl}"
           iframe-src="{environment:dvDemosBaseUrl}/menus/nav-drawer-styling"
           alt="$Platform$ Navigation Drawer Styling Example"
           github-src="menus/nav-drawer/styling">
</code-view>

## API References

* [IgcNavDrawerComponent]({environment:wcApiUrl}/classes/igcnavdrawercomponent.html)
* [IgcNavDrawerItemComponent]({environment:wcApiUrl}/classes/igcnavdraweritemcomponent.html)
* [IgcNavDrawerHeaderItemComponent]({environment:wcApiUrl}/classes/igcnavdrawerheaderitemcomponent.html)

Additional Web Components that were used:

* [IgcIconComponent]({environment:wcApiUrl}/classes/igciconcomponent.html)
* [IgcButtonComponent]({environment:wcApiUrl}/classes/igcbuttoncomponent.html)
* [IgcRadioComponent]({environment:wcApiUrl}/classes/igcradiocomponent.html)
* [IgcRadioGroupComponent]({environment:wcApiUrl}/classes/igcradiogroupcomponent.html)

## Additional Resources

<div class="divider--half"></div>

> * [Ignite UI for Web Components **GitHub**](https://github.com/IgniteUI/igniteui-webcomponents)

Reviewer 2: We already have this link below.

## Usage

> To get started with the [IgcNavigationDrawerComponent]({environment:wcApiUrl}/classes/igcnavdrawercomponent.html), first you need to install $ProductName$ by executing the following commandдддд:

Reviewer 2: ```suggestion
To get started with the [IgcNavigationDrawerComponent]({environment:wcApiUrl}/classes/igcnavdrawercomponent.html), first you need to install $ProductName$ by executing the following command:
```
